crates/ide-completion/src/item.rs

```diff
@@ -630,4 +645,19 @@ mod tests {
 
         check_relevance_score_ordered(expected_relevance_order);
     }
+
+    #[test]
+    fn exclude_non_identifier_aliases_from_lookup() {
```

Comment:
Rather than directly constructing a `CompletionItem` and verifying it, can you add an ordinary completion test in `ide-completion::tests`? It's a bit hacky but we can use [`check_edit()`] to test lookup strings don't contain non-identifier aliases.

[`check_edit()`]: https://github.com/rust-lang/rust-analyzer/blob/3eba6d30ad94fbe5146d42b9056558aa88a0dd61/crates/ide-completion/src/tests.rs#L190

Reply:
Added a `check_edit` test:

https://github.com/rust-lang/rust-analyzer/blob/4bb7702833f9c172694579adb53c830d5ba9ca2d/crates/ide-completion/src/tests/special.rs#L1284-L1307

I had to adjust `check_edit` so it doesn't check lookup strings for equality but rather for subsequence matches like LSP clients do. The test is also awkward because the tidy checks don't allow `#[should_panic]`, but this is a test that something shouldn't work, which `check_edit` handles by panicking.

Reply:
We can just pass "Partial" to (the original) `check_edit()` and see the test passes thereby asserting that the lookup string doesn't contain ">".

Reply:
That felt more brittle -- what if the lookup string changed to contain something besides just the trait name? I suppose that would break all of these tests...

Reply:
I understand your concern, but I don't think it's worth making the tests more "robust" to the changes. It's certainly annoying to break tests and have to modify every affected one, but it should be fairly straightforward and easy to reason about, and that we can easily see what and how exactly the lookup string changed is one of the advantages of being "brittle".